internal/service/certificatesigningrequest.go

```diff
@@ -165,6 +173,19 @@ func (h *ServiceHandler) CreateCertificateSigningRequest(ctx context.Context, cs
 	if errs := csr.Validate(); len(errs) > 0 {
 		return nil, api.StatusBadRequest(errors.Join(errs...).Error())
 	}
+	// Crypto validation
+	cn, ok := ctx.Value(middleware.TLSCommonNameContextKey).(string)
+
+	// Note - if auth is disabled and there is no mTLS handshake we get ok == False.
+	// We cannot check anything in that case.
+
+	if ok {
+		if csr.Spec.SignerName != "enrollment" {
+			if cn != crypto.BootstrapCNFromName(*csr.Metadata.Name) {
+				return nil, api.StatusUnauthorized(errors.New("Denied attempt to renew other entity certificate").Error())
```

Comment:
no need to create an error if you need a string?
```suggestion
				return nil, api.StatusUnauthorized("Denied attempt to renew other entity certificate")
```

Reply:
Fair point. Will do.